src/licensedcode/detection.py

```diff
@@ -118,12 +153,90 @@ class LicenseDetection:
         )
     )
 
+    matches = attr.ib(
+        #repr=matches_compact_repr,
+        default=attr.Factory(list),
+        metadata=dict(
+            help='List of license matches combined in this detection.'
+        )
+    )
+
+    @classmethod
+    def from_matches(cls, matches):
+        """
+        Return a LicenseDetection created out of `matches` list of LicenseMatch
+        objects after computing the license_expression by analyzing and combining
+        matches for the different cases.
+        """
+        reasons, license_expression = get_detected_license_expression(
+            matches=matches,
+            analysis=analyze_detection(matches),
+        )
+
+        if license_expression == None:
+            return cls(matches=matches)
+
+        spdx_license_expression = build_spdx_license_expression(
+            str(license_expression),
+            licensing=get_cache().licensing
+        )
+
+        return cls(
+            matches=matches,
+            license_expression=str(license_expression),
+            spdx_license_expression=str(spdx_license_expression),
+            combination_reasons=reasons,
+        )
+
     def __eq__(self, other):
         return (
             isinstance(other, LicenseDetection)
             and self.matches == other.matches
         )
 
+    @property
+    def query(self):
+        # A LicenseDetection will always be created with matches
+        if self.matches:
+            # All the matches in a file or in a LicenseDetection point to the
+            # same query
+            return self.matches[0].query
+    
+    @property
+    def qspans(self):
+        return [match.qspan for match in self.matches]
+
+    @property
+    def identifier(self):
+        """
+        This is an identifier for a license detection, based on it's underlying
+        license matches.
+        """
+        data = []
+        for license_match in self.original_licenses:
+            identifier = (license_match.rule_identifier, license_match.coverage(),)
+            data.append(identifier)
+
+        return tuple(data)
+    
+    @property
+    def identifier_with_text(self):
+        """
+        This is an identifier for a issue, which is an unknown license intro,
```

Comment:
Should not be mentioned. Basically in cases of small unknown matches, the matched text is required to create a unique identifier, otherwise the rule identifier and the coverage is enough in most cases. But as we discussed we need something absolute so this will change anyway.